### PR TITLE
Cleanup corrupt

### PR DESCRIPTION
**Pull request by Dignifiedquire**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -281,13 +281,25 @@ Livedb.prototype._trySubmit = function(submitData) {
...
+        return;
+      }
+
+      if (submitData.expectTransform === 2 && ops.length === 0) {
         console.warn("ERROR: CORRUPT DATA DETECTED in document "
                      + submitData.cName + '.' + submitData.docName,
                      submitData, from, to);
-        console.warn("If you're using redis, delete data for document. "
-                     + "Please file an issue if you can recreate this state reliably.");
+        console.warn("ERROR: Unable to recover");
```
Comment (joelmoss) [inline lib/index.js]: Why two `console.warn()` calls? Also, will Rollbar know about this?

Comment (Dignifiedquire) [inline lib/index.js]: we don't use rollbar on the server, we have loggly

Comment (Dignifiedquire) [inline lib/index.js]: And yes I want to leave those warnings in there like that for now.

**Comment by joelmoss** [comment]
:+1:

**Comment by destitutus** [comment]
https://github.com/codio/livedb/blob/cleanup-corrupt/lib/index.js#L47
lets use a number too

**Comment by Dignifiedquire** [comment]
> https://github.com/codio/livedb/blob/cleanup-corrupt/lib/index.js#L47
> lets use a number too

done

**Comment by destitutus** [comment]
@Dignifiedquire also I am thinking about small refactor of this method https://github.com/codio/livedb/blob/master/lib/redisdriver.js#L884
it is del only version key now, but I think about delete ops collection too. what do you think?

**Comment by Dignifiedquire** [comment]
I don't think that is needed, as this already happens (I found out when investigating this more), the redis submit script cleans up the oplog (ops collection) when something unexpected happens: https://github.com/codio/livedb/blob/master/lib/scripts/submit.lua#L19 
Also I added some logging about this and found out this corruption seems to only occur on document creation, where there is no oplog anyay.

**Comment by destitutus** [comment]
> the redis submit script cleans up the oplog

it cleanup if version is not null, but we delete version key and it will be null))

**Comment by destitutus** [comment]
but it is not critical for me

**Comment by Dignifiedquire** [comment]
> it cleanup if version is not null, but we delete version key and it will be null))

It will be non-null in the round before that, remember we only cleanup after we tried submitting twice, otherwise the conflict couldn't happen

**Comment by destitutus** [comment]
> It will be non-null in the round before that, remember we only cleanup after we tried submitting, otherwise the conflict couldn't happen

it will not happens, because setnx will return zero value (version present)

**Pull request merged**